server/routes/api/point.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+from flask import Blueprint, request, Response
+import services.datacommons as dc
+# Define blueprint
+bp = Blueprint('point', __name__, url_prefix='/api/observations/point')
+
+
+def compact_point(point_resp, all_facets):
+    result = {
+        'facets': point_resp.get('facets', {}),
+    }
+    data = {}
+    for obs_by_variable in point_resp['observationsByVariable']:
+        var = obs_by_variable['variable']
+        data[var] = {}
+        for obs_by_entity in obs_by_variable['observationsByEntity']:
+            entity = obs_by_entity['entity']
+            data[var][entity] = None
+            if 'pointsByFacet' in obs_by_entity:
+                if all_facets:
+                    data[var][entity] = obs_by_entity['pointsByFacet']
+                else:
+                    # There should be only one point.
+                    data[var][entity] = obs_by_entity['pointsByFacet'][0]
+            else:
+                if all_facets:
+                    data[var][entity] = []
+                else:
+                    data[var][entity] = {}
+    result['data'] = data
+    return result
+
+
+def point_core(entities, variables, date, all_facets):
+    resp = dc.point(entities, variables, date, all_facets)
+    return compact_point(resp, all_facets)
+
+
+def point_within_core(parent_entity, child_type, variables, date, all_facets):
+    resp = dc.point_within(parent_entity, child_type, variables, date,
+                           all_facets)
+    return compact_point(resp, all_facets)
+
+
+@bp.route('', strict_slashes=False, methods=['POST'])
+def point():
+    """Handler to get the observation point given multiple stat vars and places.
+    """
+    entities = request.json.get('entities', [])
+    variables = request.json.get('variables', [])
+    date = request.json.get('date', '')
+    return point_core(entities, variables, date, False)
+
+
+@bp.route('/all', methods=['POST'])
+def point_all():
+    """Handler to get the observation point given multiple stat vars and places.
+    """
+    entities = request.json.get('entities', [])
+    variables = request.json.get('variables', [])
+    date = request.json.get('date')
+    return point_core(entities, variables, date, True)
+
+
+@bp.route('/within', methods=['POST'])
+def point_within():
+    """Gets the statistical variable values for child entities of a certain place
```

Comment:
nit: please note in the function comment the difference between point_within and point_within_all